src/ui/settings/ViewSettingsMenu.qml

```diff
@@ -176,5 +171,38 @@ VPNFlickable {
             }
         }
     }
-}
 
+    Connections {
+        target: VPNProfileFlow
+
+        // TODO: Remove example data
+        function onPopulateFakeData() {
+            const exampleData = '{
+                "created_at": 1626704467,
+                "expires_on": 1652970067,
+                "is_cancelled": false,
+                "payment": {
+                    "credit_card_brand": "visa",
+                    "credit_card_exp_month": 12,
+                    "credit_card_exp_year": 2022,
+                    "credit_card_last4": "0016",
+                    "provider": "stripe",
+                    "type": "credit"
+                },
+                "plan": {
+                    "amount": 499,
+                    "currency": "eur",
+                    "interval_count": 1,
+                    "interval": "month"
+                },
+                "status": "active",
+                "type": "web"
+            }';
+            VPNSubscriptionData.fromJson(exampleData);
+        }
+
+        function onShowProfile() {
```

Comment:
Instead of this, you can use:
```
onStateChanged() {
  if (VPNProfileFlow.state == ready) ...
```

Reply:
Or even better, what about a state machine flow using QML states? Like in the auth?

Reply:
I think I’d like to go the `onStateChanged` route and be able to push/pop to the view stack instead of replacing view within a `Loader` here.